lib/jobs/BackupReaperJob.js

```diff
@@ -38,71 +41,122 @@ class BackupReaperJob extends BaseJob {
+        .then(() => {
+          if (numberOfBackups % 30 === 0) {
+            //Incase of many stale backups, once every 30 seconds touch the job which keeps the lock on the job
+            job.touch(() => {});
+          }
+          return Promise
+            .try(() => {
+              return backupStore.deleteBackupFile(deleteOptions, scheduledBackupOrServiceDeleted);
+            })
+            .then((response) => {
+              if (response && response === CONST.ERR_CODES.PRE_CONDITION_NOT_MET) {
+                logger.info(`${fileNameObject.backup_guid} - Backup not deleted as precondition not met`);
+                return;
+              }
+              const resp = {
+                statusCode: 200
+              };
+              const check_res_body = false;
+              eventLogger.publishAndAuditLogEvent(CONST.URL.backup_by_guid, CONST.HTTP_METHOD.DELETE, deleteOptions, resp, check_res_body);
+              logger.info(`Successfully deleted backup guid : ${fileNameObject.backup_guid}`);
+              return fileNameObject.backup_guid;
+            })
+            .catch(err => logger.error(`Error occurred while deleting backup guid: ${fileNameObject.backup_guid}. More info: `, err));
+        });
+    }
+
+    return Promise
+      .all([
+        backupStoreForInstance.listBackupFilenames(backupStartedBefore),
+        backupStoreForOob.listBackupFilenames(backupStartedBefore, {
+          root_folder: CONST.FABRIK_OUT_OF_BAND_DEPLOYMENTS.ROOT_FOLDER_NAME
+        })
+      ])
+      .spread((filesForInstance, filesForDeployments) => {
         return Promise
-          .delay(job.attrs.data.delete_delay * numberOfBackups)
-          .then(() => {
-            if (numberOfBackups % 30 === 0) {
-              //Incase of many stale backups, once every 30 seconds touch the job which keeps the lock on the job
-              job.touch(() => {});
-            }
-            return backupStore
-              .deleteBackupFile(deleteOptions, scheduledBackupOrInstanceDeleted)
-              .then((response) => {
-                if (response && response === CONST.ERR_CODES.PRE_CONDITION_NOT_MET) {
-                  logger.info(`${fileNameObject.backup_guid} - Backup not deleted as precondition not met`);
-                  return;
-                }
-                const resp = {
-                  statusCode: 200
-                };
-                const check_res_body = false;
-                eventLogger.publishAndAuditLogEvent(CONST.URL.backup_by_guid, CONST.HTTP_METHOD.DELETE, deleteOptions, resp, check_res_body);
-                logger.info(`Successfully deleted backup guid : ${fileNameObject.backup_guid}`);
-                return fileNameObject.backup_guid;
-              })
-              .catch(err => logger.error(`Error occurred while deleting backup guid: ${fileNameObject.backup_guid}. More info: `, err));
-          });
-      }).then(deletedBackupGuids => {
+          .all([
```

Comment:
since you are anyways adding a delay for each delay , should you do this in paralell or do one set of deletes after another? may be parametrize this whole method `deleteOldBackup` and call it twice? once with param 'instances' & again with param OOB? or the actual backupstores as param to this method?